astroquery/mast/cloud.py

```diff
@@ -154,8 +134,33 @@ def get_cloud_uri_list(self, data_products, include_bucket=True, full_url=False)
             List of URIs generated from the data products, list way contain entries that are None
             if data_products includes products not found in the cloud.
         """
+        s3_client = self.boto3.client('s3', config=self.config)
 
-        return [self.get_cloud_uri(product, include_bucket, full_url) for product in data_products]
+        paths = utils.mast_relative_path(data_products["dataURI"])
+        if isinstance(paths, str):  # Handle the case where only one product was requested
+            paths = [paths]
+
+        uri_list = []
+        for path in paths:
+            if path is None:
+                uri_list.append(None)
```

Comment:
Should we raise an error here like how it was done in the original `get_cloud_uri`?:
`raise InvalidQueryError("Malformed data uri {}".format(data_product['dataURI']))`

Reply:
This comment is no longer necessary.